ggshield/core/git_hooks/ci/previous_commit.py

```diff
@@ -183,6 +191,12 @@ def jenkins_previous_commit_sha(verbose: bool) -> Optional[str]:
         return get_last_commit_sha_of_branch(f"origin/{target_branch}")
 
     if previous_commit:
+        if previous_commit != EMPTY_SHA and not is_valid_git_commit_ref(
+            previous_commit
+        ):
+            if verbose:
+                click.echo("> This might be a forced push.")
+            return None
```

Comment:
Could it be EMPTY_SHA here in case of a new repo? Or would this case be handled somewhere else?

Reply:
I've not tested on Jenkins so I don't know, perhaps Jenkins never returns an EMPTY_SHA. I think you could set the line to `if push_before_sha and push_before_sha != EMPTY_SHA:` just in case, it wouldn't break anything